### ISSUE TITLE
Add support for ZFS volumes in "File Systems"

### ISSUE THREAD

**Issue by gouarfig** [OpenMediaVault-Plugin-Developers/openmediavault-zfs#5]
Hi!

I've installed the plugin from omv-extras, which is at version 2.1.0 (it looks like it might not be the latest here though?)

It works like a charm, thanks for the great work!

I just have a quick query:

When I add a volume and make a partition table in it, the partitions don't appear in the "File Systems" menu, and I have to mount them manually.

That would be nice if OMV would be able to see the /dev/zd* devices and treat them like /dev/loop* devices.

Thanks for your help

**Comment by nicjo814** [comment]
First regarding the version, it's the latest version available for OMV 2.x. I've made changes to the plugin to be compatible with OMV 3.x when it's considered stable, thus there are "newer" versions here on github, but they are not backwards compatible with OMV 2.x.

Now to the problem with the volumes. It's designed to be only one partition/filesystem on each volume (a bit of a limitation) and the intended workflow is: create volume, manually create filesystem in OMV gui, mount the filesystem in the gui. I hope this does not make it a blocker for your use.

How are /dev/loop* devices managed in OMV? Haven't really looked at those in detail.

**Comment by gouarfig** [comment]
Hi!

Thanks for your quick answer :-)

I've tried your workflow, and yes it works well that way. That's perfect for me, thanks.

In another hand, I tried the same workflow with loop devices and I get an error. I think it could be a bug in OMV (I'll check mantis if it's reported).

The way I used to make loop devices work from a disk image was to get the offset where a partition begins, and setup the loop device with that offset. Then it was appearing as a File System in OMV.

I'm not too sure which one is better, a loop device on a regular file, or a block device from a zfs volume. I suppose a block volume is specifically designed for that so it could be more efficient? or maybe it's just the same internally.

Anyway, thanks. We can close the case.
